### PR TITLE
Go port

### PR DESCRIPTION
**Pull request by progrium**
Here is the Go port with a number of changes. Most are documented in the README. It still behaves pretty identical to the original prototype with some path changes and feature additions, including a CLI. 

Try it out and give any feedback here. As long as it works mostly as good as the prototype, I'll merge into master and we'll figure out what a 0.1.0 release looks like.

### CODE REVIEW

File: scripts/enterenv [hunk 1]
```diff
@@ -1,120 +0,0 @@
...
-      --env "ENVY_SESSION=$session" \
-      --env "ENVY_RANDOM=$RANDOM" \
-      --env "ENVY_USER=$USER" \
-      --env "ENV=/env/envyrc" \
-      --env "DOCKER_HOST=unix:///env/run/docker.sock" \
-      --volume "$HOST_ROOT/.envcmd:/sbin/reload:ro" \
-      --volume "$HOST_ROOT/.envcmd:/sbin/rebuild:ro" \
-      --volume "$HOST_ROOT/.envcmd:/sbin/switch:ro" \
-      --volume "$HOST_ROOT/.envcmd:/sbin/commit:ro" \
-      --volume "$HOST_ROOT/users/$USER/envs/$ENVIRON:/env" \
```
Comment (fgrehm) [inline scripts/enterenv:63]: what happened to the `/env` dir on the go port? do we need to access it from `/envy/users/USER/environs/ENV` from now on? IMHO that seems to be a long dir path to remember :cry:

Comment (progrium) [inline scripts/enterenv:63]: /root/environ

On Friday, August 7, 2015, Fabio Rehm notifications@github.com wrote:

> In scripts/enterenv
> https://github.com/progrium/envy/pull/45#discussion_r36515236:
> 
> > -    session-reload-env "$session"
> > -    docker rm -f "$session" &> /dev/null
> > -    docker run -it \
> > -      --name "$session" \
> > -      --net "container:$USER.$ENVIRON" \
> > -      --env "HOSTNAME=$ENVIRON" \
> > -      --env "ENVY_SESSION=$session" \
> > -      --env "ENVY_RANDOM=$RANDOM" \
> > -      --env "ENVY_USER=$USER" \
> > -      --env "ENV=/env/envyrc" \
> > -      --env "DOCKER_HOST=unix:///env/run/docker.sock" \
> > -      --volume "$HOST_ROOT/.envcmd:/sbin/reload:ro" \
> > -      --volume "$HOST_ROOT/.envcmd:/sbin/rebuild:ro" \
> > -      --volume "$HOST_ROOT/.envcmd:/sbin/switch:ro" \
> > -      --volume "$HOST_ROOT/.envcmd:/sbin/commit:ro" \
> > -      --volume "$HOST_ROOT/users/$USER/envs/$ENVIRON:/env" \
> 
> what happened to the /env dir on the go port? do we need to access it
> from /envy/users/USER/environs/ENV from now on? IMHO that seems to be a
> long dir path to remember [image: :cry:]
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/progrium/envy/pull/45/files#r36515236.

## 

Jeff Lindsay
http://progrium.com

**Comment by fgrehm** [comment]
Not sure how to best solve this but I had trouble running this because my machine has Docker 1.7.1 installed and the nested DinD instance was not able to start when running with 1.6 inside the envy container.
In order to work around that I had to run envy with `-v /usr/bin/docker:/bin/docker`.

Besides that and having to figure out where my `/env` went, LGTM! Great job on this :beers:

File: scripts/enterenv (hunk 1 above)
Comment (fgrehm) [inline scripts/enterenv:63]: Oh cool, I missed out that change :blush:

**Pull request merged**